crypto/bls_multisig.go

```diff
@@ -70,16 +79,22 @@ func BLSVerifyPOP(pk PublicKey, s Signature) (bool, error) {
 // The order of the signatures in the slice does not matter since the aggregation
 // is commutative. The slice should not be empty.
 // No subgroup membership check is performed on the input signatures.
-// Expected error returns during normal operations:
-//  - invalidInputsError if no signatures are provided (sigs is empty) or
-//    at least one signature fails to deserialize.
+//
+// The function retruns:
+//  - (nil, invalidInputsError):
+//		- (nil, aggregationEmptyListError) : if no signatures are provided (input slice is empty)
+//      - Or a deserialization of at least one signature fails (input is an invalid serialization of a
+// 		compressed G1 element following [zcash]
+//     https://www.ietf.org/archive/id/draft-irtf-cfrg-pairing-friendly-curves-08.html#name-zcash-serialization-format-)
```

Comment:
From my perspective, I was thinking it is fine to check 
```
if generic_erro_type {
    if sub_error_type {} 
    else {}
}
```
If I flatten the error hierarchy and define two independent error types, we would have the same number of checks: 
```
if error_A {}
else if error_B {}
```
I agree that the second code is simpler and less prone to inserting bugs. 

Now the complicated case you described is when a caller wants to check only for the sub error type's complement,  this would look like:
```
if generic_erro_type and !sub_error_type {}
```
I agree this is not great, if we want to test for such case then we should simply add another sentinel error for these errors. I just didn't know that we needed a sentinel error case for "failing to deserialize to a BLS signature". 
I thought [you only wanted to differentiate](https://github.com/dapperlabs/flow-go/issues/6264#issuecomment-1145571653) the empty list case.

I can add a new sentinel error to cover the "failing to deserialize to a BLS signature" (I'm being careful with the name because the error doesn't only happen because of a serialization issue).